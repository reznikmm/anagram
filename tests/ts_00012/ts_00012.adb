--  Check left recursion detection

with Ada.Text_IO;

with Anagram.Grammars.Reader;
with Anagram.Grammars_Recursive_Descent;
with Anagram.Grammars_Convertors;

procedure TS_00012 is
   G : constant Anagram.Grammars.Grammar :=
     Anagram.Grammars.Reader.Read ("test.ag");

   Plain : constant Anagram.Grammars.Grammar :=
     Anagram.Grammars_Convertors.Convert (G, Left => False);

   Ok : Boolean;
begin
   Anagram.Grammars_Recursive_Descent.Generate (Plain, "-", Ok);
   Ada.Text_IO.Put_Line (Boolean'Image (Ok));
end TS_00012;
