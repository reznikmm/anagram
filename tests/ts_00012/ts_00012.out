Left recursion on non-terminal E
FALSE
